proposals/filecoin-retrieval-in-ipfs.md

```diff
@@ -0,0 +1,163 @@
+
+Future Example: Paid retrieval with IPFS & Lotus Lite client on same computer
+
+1. I make a deal with a miner through my lotus lite client to store some data. When my deal is done, the miner OR client can advertise the peer address and retrieval parameters in the IPFS dht 
+
+2. The IPFS Client sees a record for paid Filecoin retrieval in the DHT and uses the paid Filecoin retrieval plugin to talk to the Lotus Lite HTTP api to generate parameters for a paid retrieval deal. The IPFS client then initiates a data transfer request. As needed, the plugin calls out to Lotus Lite HTTP API to generate additional vouchers.
+
+Future example: Custom retrieval payment system for a "retrieval miner" using a custom payment system
+
+1. I negotiate out of band with a custom paid retrieval provider an agreement to retrieve data.
+
+2. I use the IPFS Pinning API to request to pin data. The miner stores the data in Filecoin as backup, and to prove it's being stored, but then distributes several copies to it's CDN pinning network and advertises all records in the IPFS DHT as paid retrievals that use its custom protocol.
+
+3. To retrieve, I use the information in the DHT and a plugin supplied by the provider to generate vouchers for a data transfer with one of their miners close to me. 
+
+#### Impact
+_How directly important is the outcome to web3 dev stack product-market fit?_
+
+🔥🔥🔥
+
+This would provide a clear and comprehensive story for bridging Filecoin -> IPFS. It would do so without forcing existing miners to make any changes to their operations, other than posting records in a DHT. 
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+🎯🎯🎯
+
+While the initial project would only aim to support only the free-retrieval use case, there's a clear path to all kinds of paid retrievals. It would help us flesh out key issues around discovery for retrievals. (for example extending what goes in the IPFS DHT, learning how to decide between free or paid retrieval). It should at minimum make things easier for pinning services who want to use Filecoin.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+High
+
+We know multiple companies that wish to use Filecoin as the storage provider for IPFS data. We know that many have been unsuccessful so far in bridging the gap.
+
+## Project definition
+#### Brief plan of attack
+
+1. We work with libp2p & IPFS stewards to come up with design for how we put filecoin retrieval records in the DHT. Note: I have used language so far that assumes these records will go in the IPFS DHT itself, but it may make sense to explore putting them in a seperate DHT that IPFS also has access too.
+
+2. We work out a way in Lotus to get these DHT records posted as part of the deal making process.
```

Comment:
Note: records have to be continuously reposted too, and some combination of the miner and client need to decide what records are worth posting.

Reply:
The republishing interval is a DHT-wide parameter derived from the network churn. Unless there's a separate DHT that mostly contains stable nodes, or there are enough retrieval miners to influence the properties of the public IPFS DHT it doesn't help our republishing intervals